omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleResource1_8.java

```diff
@@ -20,25 +20,36 @@
 import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
 import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
 import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Uploadable;
 import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingReadableResource;
 import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
 import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
 import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.web.WebUtil;
+import org.springframework.web.multipart.MultipartFile;
 
+import javax.servlet.ServletContext;
+import java.io.File;
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
 
-/**
- *
- */
 @Resource(name = RestConstants.VERSION_1 + "/module", supportedClass = Module.class, supportedOpenmrsVersions = { "1.8.*",
         "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
-public class ModuleResource1_8 extends BaseDelegatingReadableResource<Module> {
+public class ModuleResource1_8 extends BaseDelegatingReadableResource<Module> implements Uploadable {
+	
+	private ModuleFactoryWrapper moduleFactoryWrapper = new ModuleFactoryWrapper();
+	
+	private String moduleActionLink = ModuleActionResource1_8.class.getAnnotation(Resource.class).name();
+	
+	public void setModuleFactoryWrapper(ModuleFactoryWrapper moduleFactoryWrapper) {
+		this.moduleFactoryWrapper = moduleFactoryWrapper;
+	}
 	
 	@Override
 	public Module getByUniqueId(String uniqueId) {
-		for (Module module : ModuleFactory.getLoadedModules()) {
+		for (Module module : moduleFactoryWrapper.getLoadedModules()) {
 			if (uniqueId.equals(getUuid(module))) {
```

Comment:
I see that the ModuleResource1_8.getUuid() was added to generate a uuid on the fly using the module name, that was a mistake and may be we should fix it. UUID doesn't have to be a uuid in its literal sense, in our REST API uuid can be any unique identifier which in this case should be the module id in OpenMRS core. I think we need to use the value returned by module.getModuleId() instead of generating some arbitrary uuids.

@bmamlin  can weigh in on this